include/xrpl/protocol/detail/ledger_entries.m

```diff
@@ -0,0 +1,394 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#if !defined(LEDGER_ENTRY)
+#error "undefined macro: LEDGER_ENTRY"
```

Comment:
I just occurred to me that in order to make this file (and others) similar to a regular header, we could do this instead of `#error` :

```
#define LEDGER_ENTRY(...)
#pragma message("using dummy LEDGER_ENTRY")
```

Note the dummy macro is empty. This in turn would make the file well-behaved in IDEs and other tools, meaning it could be renamed back to `.h` if you so desire.

Reply:
@gregtatcam any comments on this ?

Reply:
Sounds good to me.

Reply:
It still doesn't solve the issue of resolving the fields reference by IDE, does it?

Reply:
You are right - it does not. An actual macro doing something could do that, maybe ?